### ISSUE TITLE
Toolchains for BELLSOFT download AdoptOpenJDK

### ISSUE THREAD

**Issue by tomaszmichalak** [gradle/gradle#17148]
### Expected Behavior
Liberica JDK is downloaded when it is specified in `java.toolchain`:

```
java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(15))
        vendor.set(JvmVendorSpec.BELLSOFT)
    }
}
```

### Current Behavior
AdoptOpenJDK is downloaded. See the log:

```
./gradlew bootRun --info
2021-05-10T14:11:56.283+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Toolchain detection' started
2021-05-10T14:11:56.288+0200 [INFO] [org.gradle.process.internal.DefaultExecHandle] Starting process 'command '/usr/libexec/java_home''. Working directory: /Users/tomasz.michalak/.gradle/daemon/6.9 Command: /usr/libexec/java_home -V
2021-05-10T14:11:56.288+0200 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: STARTING
2021-05-10T14:11:56.288+0200 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Waiting until process started: command '/usr/libexec/java_home'.
2021-05-10T14:11:56.301+0200 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: STARTED
2021-05-10T14:11:56.302+0200 [DEBUG] [org.gradle.process.internal.ExecHandleRunner] waiting until streams are handled...
2021-05-10T14:11:56.302+0200 [INFO] [org.gradle.process.internal.DefaultExecHandle] Successfully started process 'command '/usr/libexec/java_home''
2021-05-10T14:11:56.324+0200 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: SUCCEEDED
2021-05-10T14:11:56.324+0200 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Process 'command '/usr/libexec/java_home'' finished with exit value 0 (state: SUCCEEDED)
2021-05-10T14:11:56.325+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Toolchain detection'
2021-05-10T14:11:56.325+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Toolchain detection' completed
2021-05-10T14:11:56.326+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire exclusive lock on adoptopenjdk-15-x64-mac.tar.gz.
2021-05-10T14:11:56.327+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on adoptopenjdk-15-x64-mac.tar.gz.
2021-05-10T14:11:56.327+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Provisioning toolchain adoptopenjdk-15-x64-mac.tar.gz' started
2021-05-10T14:11:56.327+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Unpacking toolchain archive' started
2021-05-10T14:12:01.008+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Unpacking toolchain archive'
2021-05-10T14:12:01.009+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Unpacking toolchain archive' completed
2021-05-10T14:12:01.009+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Provisioning toolchain adoptopenjdk-15-x64-mac.tar.gz'
2021-05-10T14:12:01.009+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Provisioning toolchain adoptopenjdk-15-x64-mac.tar.gz' completed
2021-05-10T14:12:01.009+0200 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on adoptopenjdk-15-x64-mac.tar.gz.
```

### Context
I want to use Liberica JDK when creating my Spring Boot project. Spring Boot uses the Packeto buildpack support for Gradle buildinig a Docker image that uses Liberica internally. 

### Steps to Reproduce
Create a sample Spring Boot project and add:
```
java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(15))
        vendor.set(JvmVendorSpec.BELLSOFT)
    }
}
```

### Your Environment
- macOs BigSur,
- Gradle 6.9

Build scan URL: `Not required`

**Comment by tomaszmichalak** [comment]
I've checked also with Gradle 7.0. The result is the same.

**Comment by bmuskalla** [comment]
Marking as dup of #16596 for now

**Comment by mlopatkin** [comment]
Downloading the AdoptOpenJDK (actually, unpacking the already downloaded one) when the vendor is specified for the toolchain was fixed in #17509.
